File: courses/ASP.NET 2.0 - Impacta/CSWebService/App_Code/Service.cs
using System;
using System.Web;
using System.Web.Services;
using System.Web.Services.Protocols;
using System.Data;
using System.Data.SqlClient;

[WebService(Namespace = "http://tempuri.org/")]
[WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
public class Service : System.Web.Services.WebService
{
    public Service () {

        //Uncomment the following line if using designed components 
        //InitializeComponent(); 
    }

    [WebMethod]
    public string HelloWorld() {
        return "Hello World";
    }

    [WebMethod]
    public string Mensagem()
    {
        return "Isto está vindo de um Web Service.Hora:" 
              + DateTime.Now.ToString();
    }

    [WebMethod]
    public decimal Cotacao()
    {
        return Convert.ToDecimal(3.8);
    }

    [WebMethod]
    public DataSet Produtos()
    { 
        SqlDataAdapter da=new SqlDataAdapter(
            "Select ProductName, UnitPrice From Products",
            "server=localhost\\sqlexpress;database=northwind;integrated security=true;"
            );
        DataSet ds = new DataSet("loja");
        da.Fill(ds, "produto");
        return ds;
    }



    
}

File: courses/ASP.NET 2.0 - Impacta/VBWebService/App_Code/Service.vb
Imports System.Web
Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports System.Data.SqlClient
Imports System.Data


<WebService(Namespace:="http://tempuri.org/")> _
<WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _
Public Class Service
     Inherits System.Web.Services.WebService

    <WebMethod()> _
    Public Function HelloWorld() As String
        Return "Hello World"
    End Function

    <WebMethod()> Public Function Mensagem() As String
        Return "Isto está vindo de um Web Service. Hora no servidor:" _
                                      & DateTime.Now.ToString()
    End Function

    <WebMethod()> Public Function Cotacao() As Decimal
        Return Convert.ToDecimal(2.8)
    End Function

    <WebMethod()> Public Function Produtos() As DataSet
        Dim da As New SqlDataAdapter( _
          "Select ProductName, UnitPrice From Products", _
          "server=localhost\sqlexpress;database=northwind;" & _
          "integrated security=true;")
        Dim ds As New DataSet("loja")
        da.Fill(ds, "produto")
        Return ds
    End Function





End Class
